youtube_dl/PostProcessor.py

```diff
@@ -108,16 +117,55 @@ def run_ffmpeg(self, path, out_path, codec, more_opts):
 		cmd = ([self._exes['avconv'] or self._exes['ffmpeg'], '-y', '-i', encodeFilename(path), '-vn']
 			   + acodec_opts + more_opts +
 			   ['--', encodeFilename(out_path)])
-		p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-		stdout,stderr = p.communicate()
+		start = time.time()
+		# open process redirecting stderr to stdout
+		p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, close_fds=True)
+		import fcntl
+		import errno
+		import select
+		# entire captured output
+		p_output = ''
+		# size=     765kB time=243.67 bitrate=  25.7kbits/s
+		reo = re.compile("""size=\s*(?P<size>\S+)				# size
+							\stime=(?P<time>\S+)				# time
+							\sbitrate=\s*(?P<bitrate>[\d\.]+)	# bitrate
+							""", re.X)
+		# make stdout non-blocking
+		fcntl.fcntl(p.stdout.fileno(), fcntl.F_SETFL, fcntl.fcntl(p.stdout.fileno(), fcntl.F_GETFL) | os.O_NONBLOCK)
```

Comment:
Sorry for the delay, I'm looking into merging this (at least the avconv status part) just now. Why is blocking I/O insufficient here?